### PR TITLE
[CARBONDATA-2159] Remove carbon-spark dependency in store-sdk module

### PR DESCRIPTION
**Pull request by jackylk**
To make assembling JAR of store-sdk module, it should not depend on carbon-spark module
This PR changes:
1.  A `Maps` utility is added in carbon-common module to provide `getOrDefault` method and avoid JDK 8 dependency
2. `CarbonLoadModelBuilder` is added to build `CarbonLoadModel`
3. `DataLoadingUtil.scala` and `ValidateUtil.scala` is changed to java implementation and moved to `CarbonLoadModelBuilder` in processing module
4. Some common exceptions like `MalformedCarbonCommandException` are moved to carbon-common module

After all these changes, store-sdk will depend on carbon-hadoop, but not carbon-spark

 - [X] Any interfaces changed?
 No
 - [X] Any backward compatibility impacted?
 No
 - [X] Document update required?
No
 - [X] Testing done
No functionality is added 
 - [X] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA. 
NA

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/2458/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/3698/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/2459/

**Comment by jackylk** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/3699/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/2460/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/3700/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/3701/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/2461/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/2462/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/3702/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/3703/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/2463/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/2464/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/3704/

File: core/src/main/java/org/apache/carbondata/core/statusmanager/SegmentStatusManager.java [hunk 1]
```diff
@@ -708,4 +710,129 @@ public static Boolean checkIfValidLoadInProgress(AbsoluteTableIdentifier absolut
     }
   }
 
+  private static boolean isLoadDeletionRequired(String metaDataLocation) {
```
Comment (jackylk) [inline core/src/main/java/org/apache/carbondata/core/statusmanager/SegmentStatusManager.java]: These code are moved from DataLoadingUtil.scala in carbon-spark module

File: processing/src/main/java/org/apache/carbondata/processing/loading/model/CarbonLoadModelBuilder.java [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+
+  /**
+   * build CarbonLoadModel for data loading
+   * @param options Load options from user input
+   * @param optionsFinal Load options that populated with default values for optional options
+   * @param carbonLoadModel The output load model
+   * @param hadoopConf hadoopConf is needed to read CSV header if there 'fileheader' is not set in
+   *                   user provided load options
+   */
+  public void build(
```
Comment (jackylk) [inline processing/src/main/java/org/apache/carbondata/processing/loading/model/CarbonLoadModelBuilder.java]: These code are moved from DataLoadingUtil.scala in carbon-spark module

File: core/src/main/java/org/apache/carbondata/core/statusmanager/SegmentStatusManager.java [hunk 3]
```diff
@@ -708,4 +710,129 @@ public static Boolean checkIfValidLoadInProgress(AbsoluteTableIdentifier absolut
...
+    if (isLoadDeletionRequired(carbonTable.getMetadataPath())) {
+      LoadMetadataDetails[] details =
+          SegmentStatusManager.readLoadMetadata(carbonTable.getMetadataPath());
+      AbsoluteTableIdentifier identifier = carbonTable.getAbsoluteTableIdentifier();
+      ICarbonLock carbonTableStatusLock = CarbonLockFactory.getCarbonLockObj(
+          identifier, LockUsage.TABLE_STATUS_LOCK);
+
+      // Delete marked loads
+      boolean isUpdationRequired = DeleteLoadFolders.deleteLoadFoldersFromFileSystem(
+              identifier,
```
Comment (QiangCai) [inline core/src/main/java/org/apache/carbondata/core/statusmanager/SegmentStatusManager.java]: please apply java style

File: processing/src/main/java/org/apache/carbondata/processing/loading/model/LoadOption.java [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+            CarbonLoadOptionConstants.CARBON_OPTIONS_SINGLE_PASS_DEFAULT)).trim().toLowerCase();
+
+    boolean singlePass;
+
+    if (useOnePass.equalsIgnoreCase("true")) {
+      singlePass = true;
+    } else {
+      // when single_pass = false  and if either alldictionarypath
+      // or columnDict is configured the do not allow load
+      if (StringUtils.isNotEmpty(optionsFinal.get("all_dictionary_path")) || StringUtils
```
Comment (QiangCai) [inline processing/src/main/java/org/apache/carbondata/processing/loading/model/LoadOption.java]: move last StringUitls to next line

File: integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/datetype/DateTypeTest.scala [hunk 5]
```diff
@@ -16,10 +16,11 @@
  */
 package org.apache.carbondata.spark.testsuite.datetype
 
-import org.apache.carbondata.spark.exception.MalformedCarbonCommandException
 import org.apache.spark.sql.test.util.QueryTest
 import org.scalatest.BeforeAndAfterAll
 
+import org.apache.carbondata.common.exceptions.sql.MalformedCarbonCommandException
+
```
[the remark this answers is not in the document]
Comment (jackylk) [reply integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/datetype/DateTypeTest.scala]: fixed

File: core/src/main/java/org/apache/carbondata/core/statusmanager/SegmentStatusManager.java (hunk 3 above)
Comment (jackylk) [reply core/src/main/java/org/apache/carbondata/core/statusmanager/SegmentStatusManager.java]: fixed

File: processing/src/main/java/org/apache/carbondata/processing/loading/model/LoadOption.java (hunk 4 above)
Comment (jackylk) [reply processing/src/main/java/org/apache/carbondata/processing/loading/model/LoadOption.java]: fixed

**Comment by QiangCai** [comment]
LGTM

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/3713/

**Comment by jackylk** [comment]
CI failed due to environment problem in CI machine. 
merged into carbonstore branch.

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/2474/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/3516/

**Pull request closed without merging**